src/sources/prometheus/scrape.rs

```diff
@@ -152,6 +152,14 @@ impl SourceConfig for PrometheusScrapeConfig {
             endpoint_tag: self.endpoint_tag.clone(),
         };
 
+        if self.target_timeout > self.interval {
+            warn!(
+                interval_secs = %self.interval.as_secs_f64(),
+                target_timeout_secs = %self.target_timeout.as_secs_f64(),
+                message = "Having a scrape timeout that exceeds the scrape interval can lead to excessive resource consumption.",
+            );
+        }
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏this could probably be a helper function in the `util/http_client` , passing in the durations as parameters, just do avoid some duplicate code that could drift out of sync.

Reply:
fixed in https://github.com/vectordotdev/vector/pull/18021/commits/ca322e779ca047529fcef990d3bf595861bcb2c8